[Site Isolation] Find in page does not work on iOS

colelao opened the pull request:
#### df98bde355e5cd7178fb9d539bea5d9175fd2762
<pre>
[Site Isolation] Find in page does not work on iOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=302051">https://bugs.webkit.org/show_bug.cgi?id=302051</a>
<a href="https://rdar.apple.com/164130275">rdar://164130275</a>

Reviewed by Aditya Keerthi and Sihui Liu.

Before this patch, the text matches would be requested only from
one web content process. Now, we use an aggregator, and collect
all the matches from every web content process. The aggregator
then calls the completion handler in it&apos;s destructor.

* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::WebKitProtocolHandler::handleGPU):
* Source/WebKit/UIProcess/Extensions/glib/WebExtensionGLib.cpp:
* Source/WebKit/UIProcess/FindTextMatchesCallbackAggregator.cpp: Copied from Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.cpp.
(WebKit::FindTextMatchCallbackAggregator::create):
(WebKit::FindTextMatchCallbackAggregator::foundMatches):
(WebKit::FindTextMatchCallbackAggregator::~FindTextMatchCallbackAggregator):
(WebKit::FindTextMatchCallbackAggregator::FindTextMatchCallbackAggregator):
* Source/WebKit/UIProcess/FindTextMatchesCallbackAggregator.h: Copied from Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.cpp.
* Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.cpp:
* Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::findTextRangesForStringMatches):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPage.mm:
(-[TestSearchAggregator initWithCompletionHandler:]): Deleted.
(-[TestSearchAggregator foundRange:forSearchString:inDocument:]): Deleted.
(-[TestSearchAggregator finishedSearching]): Deleted.
(-[TestSearchAggregator allFoundRanges]): Deleted.
(-[TestSearchAggregator invalidateFoundRange:inDocument:]): Deleted.
(-[TestSearchAggregator invalidate]): Deleted.
(-[TestSearchAggregator count]): Deleted.
(testPerformTextSearchWithQueryStringInWebView): Deleted.
(textRangesForQueryString): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPageUtilities.h: Copied from Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.cpp.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPageUtilities.mm: Added.
(-[TestSearchAggregator initWithCompletionHandler:]):
(-[TestSearchAggregator foundRange:forSearchString:inDocument:]):
(-[TestSearchAggregator finishedSearching]):
(-[TestSearchAggregator allFoundRanges]):
(-[TestSearchAggregator invalidateFoundRange:inDocument:]):
(-[TestSearchAggregator invalidate]):
(-[TestSearchAggregator count]):
(testPerformTextSearchWithQueryStringInWebView):
(textRangesForQueryString):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/JSBuffer.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
(TestWebKitAPI::(SiteIsolation, FindStringInFrameIOS)):
(TestWebKitAPI::(SiteIsolation, FindStringInNestedFrameIOS)):
(TestWebKitAPI::(SiteIsolation, FindStringAcrossMultipleFramesIOS)):

Canonical link: <a href="https://commits.webkit.org/303197@main">https://commits.webkit.org/303197@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0da06318537a2d30ed6acbfa02ebaaccb920a47e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131488 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/3820 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42453 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/138995 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83265 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/b2d3b40b-9475-48f9-bb09-811f49b3c630) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133358 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/3844 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/3660 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/100392 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/68049 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/01f9c682-6b55-4631-9097-07ecdea35b27) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134434 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/2771 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/117691 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81184 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/d72d66df-a9e8-40ea-bc88-9e79bc09b2b4) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2684 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/427 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82187 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111303 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/35817 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141641 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/3564 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36332 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/108761 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/3612 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3180 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/108988 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27643 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2706 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114019 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/56752 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3625 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32430 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3450 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/67036 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/3716 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3555 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

colelao commented on Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
```diff
@@ -678,7 +678,7 @@ void WebKitProtocolHandler::handleGPU(WebKitURISchemeRequest* request, RenderPro
     addTableRow(displayObject, "Depth"_s, String::number(screenDepth(nullptr)));
     addTableRow(displayObject, "Bits per color component"_s, String::number(screenDepthPerComponent(nullptr)));
-    addTableRow(displayObject, "Font Scaling DPI"_s, String::number(fontDPI()));
+    addTableRow(displayObject, "Font Scaling DPI"_s, String::number(WebCore::fontDPI()));
```
This is a unified build issue resolution.

colelao commented on Source/WebKit/UIProcess/Extensions/glib/WebExtensionGLib.cpp:
```diff
@@ -30,6 +30,8 @@
 #include <wtf/FileSystem.h>
 #include <wtf/Language.h>
 #include <wtf/glib/GRefPtr.h>
+#include <wtf/text/MakeString.h>
```
This is a unified build issue resolution.

colelao commented on Source/WebKit/UIProcess/gtk/SystemSettingsManagerProxyGtk.cpp:
```diff
@@ -192,7 +192,7 @@ void SystemSettingsManagerProxy::updateFontProperties(const String& fontName, We
     // If the size of the font is in points, we need to convert it to pixels.
     if (!pango_font_description_get_size_is_absolute(pangoDescription))
-        size = size * (fontDPI() / 72.0);
+        size = size * (WebCore::fontDPI() / 72.0);
```
This is a unified build issue resolution.

szewai commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
```diff
@@ -75,6 +78,8 @@
 #endif
 
+
+
```
Unnecessary blank lines

pxlcoder replied:
```suggestion
```

pxlcoder commented on Source/WebKit/UIProcess/FindTextMatchesCallbackAggregator.cpp:
```diff
@@ -0,0 +1,55 @@
+#include "FindTextMatchesCallbackAggregator.h"
+
+#include "Shared/WebFoundTextRange.h"
+
```
```suggestion
#include "WebFoundTextRange.h"
```

szewai commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
```diff
@@ -6998,4 +7003,42 @@ HTTPServer server({
+    testPerformTextSearchWithQueryStringInWebView(webView.get(), @"Missing string", searchOptions.get(), 0UL);
+}
+
+TEST(SiteIsolation, FindStringInNestedFrameWithUIFindInteraction)
```
Could you add a test case where there are more than 1 match in different frames?

pxlcoder commented on Source/WebKit/UIProcess/FindTextMatchesCallbackAggregator.h:
```diff
@@ -0,0 +1,53 @@
+    Vector<WebFoundTextRange> m_range;
+};
+} // namespace WebKit
+
```
```suggestion
```

pxlcoder commented on Source/WebKit/UIProcess/FindTextMatchesCallbackAggregator.h:
```diff
@@ -0,0 +1,53 @@
+
+#pragma once
+
+#include "Shared/WebFoundTextRange.h"
```
```suggestion
#include "WebFoundTextRange.h"
```

pxlcoder replied:
But, can we forward declare here?

pxlcoder commented on Source/WebKit/UIProcess/FindTextMatchesCallbackAggregator.h:
```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include "Shared/WebFoundTextRange.h"
+#include <WebCore/FrameIdentifier.h>
```
Is this include necessary?

pxlcoder commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -134,6 +134,7 @@
 #include "TextChecker.h"
 #include "TextCheckerState.h"
 #include "TextRecognitionUpdateResult.h"
+#include "UIProcess/FindTextMatchesCallbackAggregator.h"
```
```suggestion
#include "FindTextMatchesCallbackAggregator.h"
```

pxlcoder commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -6305,7 +6306,16 @@ void WebPageProxy::findString(const String& string, OptionSet<FindOptions> optio
+    if (!hasRunningProcess())
+        return;
+
+    auto aggregator = FindTextMatchCallbackAggregator::create(WTFMove(callbackFunction));
```
```suggestion
    Ref aggregator = FindTextMatchCallbackAggregator::create(WTFMove(callbackFunction));
```

pxlcoder commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -6305,7 +6306,16 @@ void WebPageProxy::findString(const String& string, OptionSet<FindOptions> optio
+    auto aggregator = FindTextMatchCallbackAggregator::create(WTFMove(callbackFunction));
+
+    forEachWebContentProcess([&](auto& webProcess, auto pageID) {
+        webProcess.sendWithAsyncReply(Messages::WebPage::FindTextRangesForStringMatches(string, options, maxMatchCount), [aggregator](Vector<WebFoundTextRange>&& vector) {
```
The max match count is now per-process, but I think that's okay.

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPageUtilities.h:
```diff
@@ -0,0 +1,46 @@
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
```
```suggestion
#pragma once

```

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPageUtilities.h:
```diff
@@ -0,0 +1,46 @@
+#include <wtf/BlockPtr.h>
+#include <wtf/RetainPtr.h>
+#if PLATFORM(IOS_FAMILY)
+#import "UIKitSPIForTesting.h"
```
This header import can likely be avoided.

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
```diff
@@ -25,6 +25,9 @@
 
 #import "config.h"
 #import "DragAndDropSimulator.h"
+#if HAVE(UIFINDINTERACTION)
```
No need to guard this.

szewai commented on Source/WebKit/UIProcess/FindTextMatchesCallbackAggregator.cpp:
```diff
@@ -0,0 +1,55 @@
+    return adoptRef(*new FindTextMatchCallbackAggregator(WTFMove(completionHandler)));
+}
+
+void FindTextMatchCallbackAggregator::foundMatches(Vector<WebFoundTextRange>&& range)
```
The different web processes could reply in random order (e.g. if a nested frame is in the same process as the top-level frame, the we might get the result of the nested frame before its parent frame), so I think we will need to ensure the results are assembled in visual order (like in FindStringCallbackAggregator.cpp). 
If this is already implemented, we will need to have a test for that.

szewai replied:
If you intend to fix in followup patch let's leave a FIXME comment here.

pxlcoder commented on Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
```diff
@@ -740,9 +740,9 @@
-		A176D24D2D6D2A4E00C63915 /* WebContentExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2492D6D295400C63915 /* WebContentExtension.appex */; platformFilter = ios; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
+		A176D24B2D6D2A4E00C63915 /* GPUExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2472D6D295400C63915 /* GPUExtension.appex */; platformFilters = (ios, ); settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
+		A176D24C2D6D2A4E00C63915 /* NetworkingExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2482D6D295400C63915 /* NetworkingExtension.appex */; platformFilters = (ios, ); settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
+		A176D24D2D6D2A4E00C63915 /* WebContentExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2492D6D295400C63915 /* WebContentExtension.appex */; platformFilters = (ios, ); settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
```
Please undo.

szewai reviewed (commented):
Overall the approach looks correct

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
```diff
@@ -6998,4 +7003,42 @@ HTTPServer server({
+    [navigationDelegate waitForDidFinishNavigation];
+
+    RetainPtr searchOptions = adoptNS([[UITextSearchOptions alloc] init]);
+    testPerformTextSearchWithQueryStringInWebView(webView.get(), @"Hello world", searchOptions.get(), 1UL);
```
These test cases are a little simple. Can we test multiple matches across multiple frames?

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPage.mm:
```diff
@@ -42,6 +42,10 @@
 #import "UIKitSPIForTesting.h"
 #endif
 
+#if HAVE(UIFINDINTERACTION)
```
No need to guard this import.

szewai commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -6305,7 +6306,16 @@ void WebPageProxy::findString(const String& string, OptionSet<FindOptions> optio
 {
-    sendWithAsyncReply(Messages::WebPage::FindTextRangesForStringMatches(string, options, maxMatchCount), WTFMove(callbackFunction));
+    if (!hasRunningProcess())
+        return;
```
Still need to invoke `callbackFunction` on early return.

szewai replied:
`callbackFunction({ })` should be enough

szewai commented on Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
```diff
@@ -740,9 +740,6 @@
 		A16E590E2E7B713C0093E45F /* remove-fullscreen-element-from-dom.html in Copy Resources */ = {isa = PBXBuildFile; fileRef = A16E590D2E7B71310093E45F /* remove-fullscreen-element-from-dom.html */; };
 		A170E3AA2ABA7857009EA799 /* VectorCocoa.mm in Sources */ = {isa = PBXBuildFile; fileRef = A170E3A22ABA7857009EA799 /* VectorCocoa.mm */; };
 		A17191AD2CD2DB090088944E /* WKWebViewLogging.mm in Sources */ = {isa = PBXBuildFile; fileRef = A17191AC2CD2DB090088944E /* WKWebViewLogging.mm */; };
-		A176D24B2D6D2A4E00C63915 /* GPUExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2472D6D295400C63915 /* GPUExtension.appex */; platformFilter = ios; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
```
Why this change?

szewai commented on Source/WebKit/UIProcess/FindTextMatchesCallbackAggregator.h:
```diff
@@ -0,0 +1,49 @@
+
+private:
+    FindTextMatchCallbackAggregator(CompletionHandler<void(Vector<WebFoundTextRange>&&)>&&);
+    CompletionHandler<void(Vector<WebFoundTextRange>)> m_completionHandler;
```
`CompletionHandler<void(Vector<WebFoundTextRange>&&)>`

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPage.mm:
```diff
@@ -42,6 +42,8 @@
 #import "UIKitSPIForTesting.h"
 #endif
 
+#import "FindInPageUtilities.h"
```
Move up with the other imports.

szewai commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPage.mm:
```diff
@@ -42,6 +43,7 @@
 #import "UIKitSPIForTesting.h"
 #endif
 
+
```
Remove?

colelao committed:
[Site Isolation] Find in page does not work on iOS
https://bugs.webkit.org/show_bug.cgi?id=302051
rdar://164130275

Reviewed by Aditya Keerthi and Sihui Liu.

Before this patch, the text matches would be requested only from
one web content process. Now, we use an aggregator, and collect
all the matches from every web content process. The aggregator
then calls the completion handler in it's destructor.

* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::WebKitProtocolHandler::handleGPU):
* Source/WebKit/UIProcess/Extensions/glib/WebExtensionGLib.cpp:
* Source/WebKit/UIProcess/FindTextMatchesCallbackAggregator.cpp: Copied from Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.cpp.
(WebKit::FindTextMatchCallbackAggregator::create):
(WebKit::FindTextMatchCallbackAggregator::foundMatches):
(WebKit::FindTextMatchCallbackAggregator::~FindTextMatchCallbackAggregator):
(WebKit::FindTextMatchCallbackAggregator::FindTextMatchCallbackAggregator):
* Source/WebKit/UIProcess/FindTextMatchesCallbackAggregator.h: Copied from Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.cpp.
* Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.cpp:
* Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::findTextRangesForStringMatches):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPage.mm:
(-[TestSearchAggregator initWithCompletionHandler:]): Deleted.
(-[TestSearchAggregator foundRange:forSearchString:inDocument:]): Deleted.
(-[TestSearchAggregator finishedSearching]): Deleted.
(-[TestSearchAggregator allFoundRanges]): Deleted.
(-[TestSearchAggregator invalidateFoundRange:inDocument:]): Deleted.
(-[TestSearchAggregator invalidate]): Deleted.
(-[TestSearchAggregator count]): Deleted.
(testPerformTextSearchWithQueryStringInWebView): Deleted.
(textRangesForQueryString): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPageUtilities.h: Copied from Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.cpp.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/FindInPageUtilities.mm: Added.
(-[TestSearchAggregator initWithCompletionHandler:]):
(-[TestSearchAggregator foundRange:forSearchString:inDocument:]):
(-[TestSearchAggregator finishedSearching]):
(-[TestSearchAggregator allFoundRanges]):
(-[TestSearchAggregator invalidateFoundRange:inDocument:]):
(-[TestSearchAggregator invalidate]):
(-[TestSearchAggregator count]):
(testPerformTextSearchWithQueryStringInWebView):
(textRangesForQueryString):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/JSBuffer.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm:
(TestWebKitAPI::(SiteIsolation, FindStringInFrameIOS)):
(TestWebKitAPI::(SiteIsolation, FindStringInNestedFrameIOS)):
(TestWebKitAPI::(SiteIsolation, FindStringAcrossMultipleFramesIOS)):

Canonical link: https://commits.webkit.org/303197@main
